dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -232,6 +232,103 @@ void dpnp_modf_c(void* array1_in, void* result1_out, void* result2_out, size_t s
+                                                                   _DataType_output>>(gws, kernel_parallel_for_func);
+    };
+    event = DPNP_QUEUE.submit(kernel_func);
+    event.wait();
+}
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+class dpnp_multiply_scalar_array_c_kernel;
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_multiply_scalar_array_c(void* input1_in, void* input2_in, void* result_out, const size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+
+    cl::sycl::event event;
+    _DataType_input1* input1_ptr = reinterpret_cast<_DataType_input1*>(input1_in);
+    _DataType_input1 input1 = *input1_ptr;
+    _DataType_input2* input2 = reinterpret_cast<_DataType_input2*>(input2_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result_out);
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0]; /*for (size_t i = 0; i < size; ++i)*/
+        {
+            result[i] = input1 * input2[i];
+        }
+    };
+    auto kernel_func = [&](cl::sycl::handler& cgh) {
+        cgh.parallel_for<class dpnp_multiply_scalar_array_c_kernel<_DataType_input1, _DataType_input2,
+                                                                   _DataType_output>>(gws, kernel_parallel_for_func);
+    };
+    event = DPNP_QUEUE.submit(kernel_func);
+    event.wait();
+}
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_multiply_c(void* input1_in,
+                     void* input2_in,
+                     void* result_out,
+                     const size_t input1_size,
+                     const size_t input2_size)
+                    //  long* input1_shape,
+                    //  long* input2_shape,
+                    //  const size_t input1_shape_ndim,
+                    //  const size_t input2_shape_ndim)
+{
+    if (!input1_size || !input2_size)
+    {
+        return;
+    }
+
+    if (input1_size == input2_size)
+    {
+        dpnp_multiply_array_array_c<_DataType_input1, _DataType_input2, _DataType_output>(input1_in, input2_in,
+                                                                                          result_out, input1_size);
+    }
+    else if (input1_size == 1)
+    {
+        dpnp_multiply_scalar_array_c<_DataType_input1, _DataType_input2, _DataType_output>(input1_in, input2_in,
+                                                                                           result_out, input2_size);
+    }
+    else if (input2_size == 1)
+    {
+        dpnp_multiply_array_scalar_c<_DataType_input1, _DataType_input2, _DataType_output>(input1_in, input2_in,
+                                                                                           result_out, input1_size);
```

Comment:
Actually, you don't need different funcs for `array_vs_scalar` and `scalar_vs_array` cases, just reuse one of them.